egs/wsj/s5/steps/dur_model/train.sh

```diff
@@ -0,0 +1,156 @@
+   echo "  --minibatch-size <size>                     # minibatch size"
+   echo "  --compute-prob-interval <int>               # interval for measuring accuracy (diagnostics)"
+   echo "  --nnet-config <nnet3-conf-file>             # use this config for training"
+   echo "Options related to initializing (stage 0):"
+   echo "  --max-duration <duration-in-frames>         # max duration; if not set, it will be determined automatically"
+   echo "  --left-context <size>                       # left phone context size"
+   echo "  --right-context <size>                      # right phone context size"
+   exit 1;
+fi
+
+phones_dir=$1
+alidir=$2
+dir=$3
+
+durmodel=$dir/durmodel.mdl
+mkdir -p $dir/log
+
+if [ $stage -le 0 ]; then
+  echo "$0: Initializing the duration model and nnet..."
+
+  transmodel=$alidir/final.mdl
+  for f in $transmodel $phones_dir/roots.int $phones_dir/extra_questions.int; do
+    [ ! -f $f ] && echo "$0: Required file for initializing not found: $f" && exit 1;
+  done
+
+  if [ $max_duration == 0 ]; then
+    echo "$0: Determining max-duration..."
+    if [ ! -f $dir/all.ali ]; then
+      gunzip -c $alidir/ali.*.gz >> $dir/all.ali || exit 1;
+    fi
+
+    max_duration=`ali-to-phones --write-lengths $transmodel ark:$dir/all.ali \
+                  ark,t:- | awk -v min_count=$min_repeat_count -F';' \
+                  '{ for(i=1; i<=NF; i++){ \
+                       n=split($(i), a, " ");\
+                       duration=a[n];\
+                       counts[duration]++;\
+                       if(duration>max_duration) max_duration=duration; \
+                   }} END { for(d=3; d<=max_duration; d++) \
+                              if(counts[d]<min_count){\
+                                print d-1; exit 0;\
+                              }\
+                            print max_duration;\
+                            exit 0;}'` || exit 1;
+
+  fi
+  echo "$0: Max duration: $max_duration"
+
+  $cmd $dir/log/durmod_init.log \
+       durmodel-init --max-duration=$max_duration \
+                     --left-context=$left_context \
+                     --right-context=$right_context \
+                     $phones_dir/roots.int $phones_dir/extra_questions.int \
+                     $durmodel || exit 1;
+
+fi
+
+if [ $stage -le 1 ]; then
+  echo "$0: Preparing examples..."
+  steps/dur_model/get_examples.sh --cmd $cmd $egs_opts $alidir $dir || exit 1;
+fi
+
+
+if [ $stage -le 2 ]; then
+
+  [ ! -f $durmodel ] && echo "$0: Duration model file not found (have you completed stage 0?): $durmodel" && exit 1;
+  [ ! -f $dir/train.egs ] && echo "$0: Train examples file not found (have you completed stage 1?): $dir/train.egs" && exit 1;
+  [ ! -f $dir/val.egs ] && echo "$0: Validation examples file not found (have you completed stage 1?): $dir/val.egs" && exit 1;
+
+  if $use_gpu; then
+    train_queue_opt="--gpu 1"
+    if [ $cmd != "queue.pl" ]; then
```

Comment:
I think you can remove this warning and the 'read' statement-- user interaction is not something people really expect in Kaldi scripts.  Anyway the test wouldn't work if options were given to queue.pl, which they very often are in $cmd strings.